ver 0.51

Darlitch opened the pull request.

Darlitch committed:
Create .gitignore

Darlitch committed:
ver 0.1 create project

Darlitch committed:
ver 0.2 Create structure of project and Exception

Darlitch committed:
ver 0.3 UserSearchService is done. Add ErrorDTO and DTO for UserSearchService.cs

Darlitch committed:
ver 0.4 UserSearch is done?

Darlitch committed:
Update .gitignore

Darlitch committed:
ver 0.5 fix all bugs. User Search has been tested

Darlitch committed:
ver 0.51 fix structure of project

Darlitch committed:
fix(structure)

Create new dockerfile and fix .gitignore

Darlitch committed:
fix(structure)

Change structure: delete .idea from github

Darlitch committed:
fix(structure)

Restore dockerfile from project Api. Add dockerfile to sln

Darlitch committed:
fix(structure)

Restore dockerfile from project Api. Add dockerfile to sln

Darlitch committed:
build(core): finalize project structure for version 0.85

Create ERD, project, migrations, triggers, models, DTOs, repos, services, controllers, validation, Docker, and fix bugs.

Darlitch committed:
build(core): finalize project for version 0.55

Service, Manager and Repository for User Search Done. All bugs are fixed

Pauk456 commented on Search.Service/Service/UserSearchService.cs:
```diff
@@ -0,0 +1,49 @@
+    public async Task<PagedUsersMainDTO> GetUsersByNicknameAsync(string nickname, int offset, int limit,
+        CancellationToken ct = default)
+    {
+        var response = await httpClient.GetAsync($"/api/users/search-api?nickname={Uri.EscapeDataString(nickname)}&offset={offset}&limit={limit}", ct);
```
Все url в отдельный Const класс

Pauk456 commented on Search.Service/Manager/UserSearchManager.cs:
```diff
@@ -0,0 +1,28 @@
+        return await userSearchService.GetUserByUidAsync(uid, ct);
+    }
+
+    public async Task<PagedUsersMainDTO> SearchUsersByNicknameAsync(string nickname, int offset, int limit,
```
А че мы решили по поводу чувствительности к регистру

Darlitch replied:
Регистр важен!

Darlitch committed:
ver 0.851 add swagger

Darlitch committed:
refactor: fix issues and add request factory for external requests

Updated search service structure by applying all review fixes and improving URL construction.
Introduced RequestFactory and RequestPath to centralize request building logic and remove duplication.
